mypy/expandtype.py

```diff
@@ -441,8 +445,19 @@ def expand_types_with_unpack(self, typs: Sequence[Type]) -> list[Type]:
                 items.append(item.accept(self))
         return items
 
+    def expand_type_tuple_with_unpack(self, typs: tuple[Type, ...]) -> list[Type]:
```

Comment:
Huh, it's an interesting optimization I'd never think of. Is this difference significant because of some list operations special-cased by mypyc (something like bypassing `__getitem__` and dispatching to `PyList_GetItem` directly)?

Reply:
Yeah, mypyc uses specialized primitives for the most common built-in types, including `list` and `tuple`. @jhance  has a WIP implementation of a faster sequence iteration implementation, but narrower types are still going to be at least a little more efficient. The specialized operations are listed in the docs -- here are the ones for `list`: https://mypyc.readthedocs.io/en/latest/list_operations.html